src/uct/ib/base/ib_iface.c

```diff
@@ -479,42 +479,37 @@ void uct_ib_address_unpack(const uct_ib_address_t *ib_addr,
 
         /* If the link layer is not ETHERNET, then it is IB and a lid
          * must be present */
-        params.lid                      = *(const uint16_t*)ptr;
-        ptr                             = UCS_PTR_TYPE_OFFSET(ptr, uint16_t);
+        params.lid = ucs_serialize_get(ptr, uint16_t);
```

Comment:
the same to get:
`ptr = ucs_serialize_get(ptr, &value)`/`ptr = ucs_serialize_typed_get(ptr, type, &value)`

Reply:
it was done similarly to "va_arg" logic, and to avoid using "ptr" twice. maybe pass &ptr for clarity?
also, IMO better pass type, because sometimes the type of "value" variable is not exactly the same as the packed type (e.g enum path_mtu)

Reply:
> it was done similarly to "va_arg" logic

I would try to avoid macro especially with side effects, why we can not use pure functions here?

>  IMO better pass type, because sometimes the type of "value" variable is not exactly the same as the packed type

then maybe pass exact length?